File: java/58-length-of-last-word.java
class Solution{

    // first solution
    public int lengthOfLastWord1(String s) {

        s = s.trim(); // remove leading and trailing spaces
        int length = 0;

        // iterate from the end of the string
        for (int i = s.length() - 1; i >= 0; i--) {
            if (s.charAt(i) != ' ') {
                length++;
            } else if (length > 0) {
                break;
            }
        }

        return length;
    }

    // second solution more straightforward logic and readable
    public int lengthOfLastWord(String s) {

        s = s.trim(); // remove leading and trailing spaces
        int length = 0;

        // iterate from the end of the string
        for (int i = s.length() - 1; i >= 0; i--) {
            if (s.charAt(i) == ' ') {
                break;
            }
            length++;
        }

        return length;
    }
}

File: go/58-length-of-last-word.go
func lengthOfLastWord(s string) int {

    s = strings.TrimSpace(s) // remove leading and trailing spaces
    length := 0

    // iterate from the end of the string
    for i := len(s) - 1; i >= 0; i-- {
        if s[i] == ' ' {
            break
        }
        length++
    }

    return length
}